simln-lib/src/lib.rs

```diff
@@ -36,13 +36,17 @@ pub mod sim_node;
 #[cfg(test)]
 mod test_utils;
 
+/// Represents a node id, either by its public key or alias.
 #[derive(Serialize, Debug, Clone)]
 pub enum NodeId {
+    /// The node's public key
     PublicKey(PublicKey),
+    /// The node's alias (human-readable name)
     Alias(String),
 }
 
 impl NodeId {
+    /// Validates that the provided node id matches the one returned by the backend.
```

Comment:
Not strictly true, because we allow aliases that don't match - perhaps make a note of this?